server/datastore/mysql/labels.go

```diff
@@ -615,12 +615,13 @@ func (ds *Datastore) applyHostLabelFilters(ctx context.Context, filter fleet.Tea
 	// 	// TODO: Do we currently support filtering by software version ID and label?
 	// }
 	if opt.SoftwareTitleIDFilter != nil && opt.SoftwareStatusFilter != nil {
-		// get the installer id
-		meta, err := ds.GetSoftwareInstallerMetadataByTeamAndTitleID(ctx, opt.TeamFilter, *opt.SoftwareTitleIDFilter, false)
+		// check for software installer metadata
+		_, err := ds.GetSoftwareInstallerMetadataByTeamAndTitleID(ctx, opt.TeamFilter, *opt.SoftwareTitleIDFilter, false)
 		if err != nil {
 			return "", nil, ctxerr.Wrap(ctx, err, "get software installer metadata by team and title id")
 		}
-		installerJoin, installerParams, err := ds.softwareInstallerJoin(meta.InstallerID, *opt.SoftwareStatusFilter)
+		// TODO(sarah): are we missing VPP apps here? see ds.applyHostFilters
+		installerJoin, installerParams, err := ds.softwareInstallerJoin(*opt.SoftwareTitleIDFilter, *opt.SoftwareStatusFilter)
```

Comment:
Seems like you're right, in applyHostFilters it calls `vppAppJoin`: https://github.com/fleetdm/fleet/blob/942670105f143f2f39280d971fdd4002a35b3a60/server/datastore/mysql/hosts.go#L1099

I'm assuming you'll tackle VPP status filter in another PR?